spec/lib/active_versioning/nested_attributes_spec.rb

```diff
@@ -0,0 +1,119 @@
+require 'spec_helper'
+
+RSpec.describe "assigning nested attributes" do
+  let!(:post) do
+    ActiveVersioning::Test::Post.create(
+      title: 'So Post',
+      body: 'Such interesting. Very wow.'
+    )
+  end
+
+  let!(:committed_at) { Time.utc(2014, 11, 26, 12, 42, 35) }
+
+  let(:subject) { post.current_draft }
+
+  it "creates a new belongs-to relationship" do
+    expect {
+      subject.assign_attributes(author_attributes: { name: "Steve" })
+      subject.save
+    }.not_to change {
+      ActiveVersioning::Test::User.count
+    }
+
+    p = ActiveVersioning::Test::Post.find(post.id).current_draft
```

Comment:
I do this over and over to ensure that the object is really, truly reloaded. Just calling `.reload` isn't always enough given the proxy/delegator system in place.